psh/test-date.py

```diff
@@ -0,0 +1,154 @@
+    psh.assert_cmd(p, "date -s 123456789operand1", expected=invalid_date("123456789operand1"), result="fail")
+
+    # too many arguments
+    psh.assert_cmd(p, "date -s 1600000000 +format operand1", expected=unrec_arg("operand1"), result="fail")
+    psh.assert_cmd(p, "date", expected=DFLT_DATE_REG, is_regex=True)
+
+
+def test_incorrect_dateparse(p):
+    # No argument passed
+    psh.assert_cmd(p, "date -d", expected="date: option requires an argument -- d", result="fail")
+    psh.assert_cmd(p, "date", expected=DFLT_DATE_REG, is_regex=True)
+
+    # too many arguments
+    psh.assert_cmd(p, "date -d @1600000000 +format operand1", expected=unrec_arg("operand1"), result="fail")
+    psh.assert_cmd(p, "date", expected=DFLT_DATE_REG, is_regex=True)
+
+    # invalid time value
+    psh.assert_cmd(p, "date -d @1600000000operand1", expected=invalid_date("@1600000000operand1"), result="fail")
+    psh.assert_cmd(p, "date", expected=DFLT_DATE_REG, is_regex=True)
+
+    # no '@' sign
+    psh.assert_cmd(p, "date -d 1600000000", expected=invalid_date("1600000000"), result="fail")
+    psh.assert_cmd(p, "date", expected=DFLT_DATE_REG, is_regex=True)
+
+
+def test_edges(p):
+    """Test edge cases for epoch value passed to `date -s epoch"""
+
+    # maximum value for int32
+    psh.assert_cmd(p, "date -s @2147483647", expected=r"Tue, 19 Jan 38 \d{2}:\d{2}:\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date", expected=r"Tue, 19 Jan 38 \d{2}:\d{2}:\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date +%Y", expected="2038")
+    psh.assert_cmd(p, "date -s @0", expected=DFLT_DATE_REG, is_regex=True)
+
+    psh.assert_cmd(p, "date -d @2147483647", expected=r"Tue, 19 Jan 38 \d{2}:\d{2}:\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date", expected=DFLT_DATE_REG, is_regex=True)
+
+    # beyond int32 value
+    psh.assert_cmd(p, "date -s @2147483648", expected=r"Tue, 19 Jan 38 \d{2}:\d{2}:\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date", expected=r"Tue, 19 Jan 38 \d{2}:\d{2}:\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date +%Y", expected="2038")
+    psh.assert_cmd(p, "date -s @0", expected=DFLT_DATE_REG, is_regex=True)
+
+    psh.assert_cmd(p, "date -d @2147483648", expected=r"Tue, 19 Jan 38 \d{2}:\d{2}:\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date", expected=DFLT_DATE_REG, is_regex=True)
+
+    # Issue: #370 https://github.com/phoenix-rtos/phoenix-rtos-project/issues/370
+    return
+
+    psh.assert_cmd(p, "date -s @10000000000000000", expected=r"Sat, 25 Jan 55 \d{2}:\d{2}:\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date", expected=r"Sat, 25 Jan 55 \d{2}:\d{2}:\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date +%Y", expected="316889355")
+    psh.assert_cmd(p, "date -s @0", DFLT_DATE_REG, is_regex=True)
+
+    psh.assert_cmd(p, "date -d @10000000000000000", expected=r"Sat, 25 Jan 55 \d{2}:\d{2}:\d{2}\r+\n", is_regex=True)
+    psh.assert_cmd(p, "date", expected=DFLT_DATE_REG, is_regex=True)
+
+
```

Comment:
Why checks like that were removed?
 - `psh.assert_cmd(p, "date -s @1600000000.1234567890", expected=SPEC_DATE_REG, is_regex=True)`